app/src/org/commcare/utils/StorageUtils.java

```diff
@@ -63,9 +62,10 @@ public static int getNumIncompleteForms() {
             return new FormRecord[0];
         }
 
-        // Order ids so they're submitted to and processed by the server in
-        // the correct order.
-        sortRecordsByDate(ids, storage);
+        if (sortRequired) {
+            // Order ids so they're submitted to and processed by the server in the correct order.
+            sortRecordsBySubmissionOrderingNumber(ids, storage);
```

Comment:
Ah gotcha, I see now! You're right, I have no idea why it was like that, I was just blindly maintaining the code structure that had been there when we were sorting by last modified date, but it does seem silly. I just changed this in the most recent commit (2df14d2). @ctsims Let me know if there was a reason for it being done the old way, but I'm guessing it was just a result of what util methods already existed when someone went to write that code, and nothing more than that.